torch/_export/passes/move_to_device_pass.py

```diff
@@ -0,0 +1,81 @@
+from typing import Dict, Union
+
+import torch
+from torch.export import ExportedProgram
+
+
+def _move_nn_param(
+    v: torch.nn.Parameter, location: Union[torch.device, str, Dict[str, str]]
+) -> torch.nn.Parameter:
+    if isinstance(location, dict):
+        if str(v.device) in location.keys():
+            return torch.nn.Parameter(v.to(location[str(v.device)]))
+        else:
+            raise KeyError(f"Device {str(v.device)} not found in location dict")
+    else:
+        return torch.nn.Parameter(v.to(location))
+
+
+def _move_tensor(
+    v: torch.Tensor, location: Union[torch.device, str, Dict[str, str]]
+) -> torch.Tensor:
+    if isinstance(location, dict):
+        if str(v.device) in location.keys():
+            return v.to(location[str(v.device)])
+        else:
+            raise KeyError(f"Device {str(v.device)} not found in location dict")
+    else:
+        return v.to(location)
+
+
+def _move_kwargs(
+    kwargs: Dict[str, object], location: Union[torch.device, str, Dict[str, str]]
+) -> Dict[str, object]:
+    new_kwargs = kwargs.copy()
+    if isinstance(location, dict):
```

Comment:
what if we just have a function which does the "if isinstance(location, dict)" logic. There seems to be a lot of duplicated logic between the 3 functions.

Reply:
I implemented a `_move_by_dict` helper function that handles the case for `torch.Tensor` and `torch.nn.Parameter`. It doesn't handle `kwargs` as I think that will make this helper function less neat. So the implementation of `_move_kwargs` didn't change.

Reply:
I feel like the code could be largely simplified if you just make a function like:
```python
def get_new_device(curr_device: str):
    if isinstance(location, dict):
        return location.get(str(curr_device), curr_device)
    else:
        return location

...

# move parameter
state_dict[k] = torch.nn.Parameter(v.to(get_new_device(v.device, location)))
# move tensor
v.to(get_new_device(v.device, location))
# move kwarg
if "device" in node.kwargs:
    node.kwargs["device"] = get_new_device(node.kwargs["device"], location)
```